Newsletters: add 153 (2021-06-16) (#590)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2021-06-16-newsletter.md

---
title: 'Bitcoin Optech Newsletter #153'
permalink: /en/newsletters/2021/06/16/
name: 2021-06-16-newsletter
slug: 2021-06-16-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter celebrates the lock in of the taproot soft fork,
describes a draft BIP for improving transaction
privacy by varying the fields used to implement anti fee sniping, and
features an article about the challenges of combining transaction
replacement with payment batching.  Also included are our regular
sections with announcements of new software releases and release
candidates, plus notable changes to popular Bitcoin infrastructure
software.

## News

- **🟩  Taproot locked in:** the [taproot][topic taproot] soft fork and
  related changes specified in BIPs [340][bip340], [341][bip341], and
  [342][bip342] were locked in by signaling miners last weekend.
  Taproot will be safe to use after block 709,632, which is expected in
  early or mid November.  The delay gives time for users to upgrade

>   their nodes to a release (such as Bitcoin Core 0.21.1) that will

Reviewer: Or just clarify that 0.21.1 is the first version that enforces

  enforce taproot's rules, ensuring that funds received to taproot
  scripts after block 709,632 are safe even if there's a problem with
  miners.

    Developers are encouraged to start [implementing taproot][taproot
    uses] so they can
    be ready to take advantage of greater efficiency, privacy, and
    fungibility as soon as the activation is complete.

    Readers celebrating the lock in of taproot may also wish to read a

>     [short history][wuille taproot] of taproot's origins and history by

Reviewer: Perhaps
```suggestion
    [short recount][wuille taproot] of taproot's origins and history by
```

    developer Pieter Wuille.

- **BIP proposed for wallets to set nSequence by default on taproot transactions:**
  Chris Belcher [posted][belcher post] a draft BIP to the Bitcoin-Dev
  mailing list suggesting an alternative way wallets can implement [anti
  fee sniping][topic fee sniping].  The alternative method would enhance
  the privacy and fungibility of transactions made by single-sig users,
  [multisignature][topic multisignature] users, and users of certain
  contract protocols such as taproot-enabled LN or advanced
  [coinswaps][topic coinswap].

    Anti fee sniping is a technique some wallets implement to discourage
    miners from trying to steal fees from each other in a way that would
    reduce the amount of proof of work expended on securing Bitcoin and

>     limit users' ability to rely on confirmation scores.  All wallets

Reviewer: Perhaps
```suggestion
    limit users' ability to rely on initial confirmations.  All wallets
```

Author: The examples here focus on a single block reorg, but it can get much much worse.

    allow users of regular single-sig transactions or uncomplicated
    multisignatures to join together with users of contract protocols to
    mutually improve each others' privacy and fungibility.

## FIXME:article

FIXME:harding to rebase on #582 on Tuesday

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [Rust Bitcoin 0.26.2][] is the project's latest minor release.
  Compared to the previous major version, it contains a several API
  improvements and bug fixes.  See the [changelog][rb changelog] for
  details.

- [Rust-Lightning 0.0.98][] is a minor release containing several
  improvements and bug fixes.  <!-- there's no release notes or
  changelog I can see, so not much to say here. -->

- [LND 0.13.0-beta.rc5][LND 0.13.0-beta] is a release candidate that
  adds support for using a pruned Bitcoin full node, allows receiving
  and sending payments using Atomic MultiPath ([AMP][topic multipath payments]),
  and increases its [PSBT][topic psbt] capabilities, among other improvements

>   and bug fixes.

Author: Also included in the week before that.  I don't usually change the text between RCs unless something notable happens.  In this case, they didn't even change the RC number (which is fine IMO, especially since I think it's partly due to several people being in Miami).

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
repo], [Hardware Wallet Interface (HWI)][hwi repo],
[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
BOLTs][bolts repo].*

- [Bitcoin Core GUI #4][] adds initial support for using [Hardware Wallet
  Interface (HWI)][topic hwi] external signers via the GUI. Once this feature is
  finalized, users will be able to use their HWI-compatible hardware wallets
  directly from the Bitcoin Core GUI.

    {:.center}
    ![Screenshot of HWI path configuration option](/img/posts/2021-06-gui-hwi.png)

- [Bitcoin Core #21573][] updates the version of libsecp256k1 included
  in Bitcoin Core.  The most notable change is the use of the
  optimized modular inverse code described in Newsletters [#136][news136
  safegcd] and [#146][news146 safegcd].  Performance evaluations posted

>   to the PR found it to speed up old block verification by about 10%.

Reviewer: Infinitive looks good to me here

  their nodes to a release (such as Bitcoin Core 0.21.1) that will
  enforce taproot's rules, ensuring that funds received to taproot
  scripts after block 709,632 are safe even if there's a problem with
  miners.

    Developers are encouraged to start [implementing taproot][taproot
    uses] in their software during the next several months so they can
    be ready to take advantage of greater efficiency, privacy, and
    fungibility as soon as the activation is complete.

    Readers celebrating the lock in of taproot may also wish to read a
    [short history][wuille taproot] of taproot's origins and history by
    developer Pieter Wuille.

- **BIP proposed for wallets to set nSequence by default on taproot transactions:**
  Chris Belcher [posted][belcher post] a draft BIP to the Bitcoin-Dev
  mailing list suggesting an alternative way wallets can implement anti
  fee sniping that would enhance the privacy and fungibility of
  transactions made by single-sig users, [multisignature][topic
  multisignature] users, and users of certain contract protocols such as
  taproot-enabled LN or advanced [coinswaps][topic coinswap].

>     Fee sniping is a problem that may occur as Bitcoin's subsidy

Author: Do you mean for me to do that in this PR (not a problem; the text is already written and we already have anti fee sniping on our topic todo), or do you mean for that to be done separately?

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- FIXME:harding Rust-Bitcoin 0.26.1 and .2

- FIXME:harding Rust-Bitcoin 0.0.98

- [LND 0.13.0-beta.rc5][LND 0.13.0-beta] is a release candidate that
  adds support for using a pruned Bitcoin full node, allows receiving
  and sending payments using Atomic MultiPath ([AMP][topic multipath payments]),
  and increases its [PSBT][topic psbt] capabilities, among other improvements
  and bug fixes.

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
repo], [Hardware Wallet Interface (HWI)][hwi repo],
[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
BOLTs][bolts repo].*

- [Bitcoin Core GUI #4][] UI external signer support (e.g. hardware wallet) FIXME:dongcarl

- [Bitcoin Core #21573][] updates the version of libsecp256k1 included

>   in Bitcoin Core.  Of the changes, the most notable in the use of the
>   optimized modular inverse code described in Newsletters [#136][news136

Reviewer 2: ```suggestion
  in Bitcoin Core.  Of the changes, the most notable is the use of the
  optimized modular inverse code described in Newsletters [#136][news136
```

Or simplify the sentence further:

```suggestion
  in Bitcoin Core.  The most notable change is the use of the
  optimized modular inverse code described in Newsletters [#136][news136
```

---
title: 'Bitcoin Optech Newsletter #153'
permalink: /en/newsletters/2021/06/16/
name: 2021-06-16-newsletter
slug: 2021-06-16-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter celebrates the lock in of the taproot soft fork,
describes a draft BIP for improving transaction
privacy by varying the fields used to implement anti fee sniping, and

> features an article about the challenges of mixing transaction

Reviewer: Maybe:

```suggestion
features an article about the challenges of combining transaction
```

replacement with payment batching.  Also included are our regular
sections with announcements of new software releases and release
candidates, plus notable changes to popular Bitcoin infrastructure
software.

## News

- **🟩  Taproot locked in:** the [taproot][topic taproot] soft fork and
  related changes specified in BIPs [340][bip340], [341][bip341], and
  [342][bip342] were locked in by signaling miners last weekend.
  Taproot will be safe to use after block 709,632, which is expected in
  early or mid November.  The delay gives time for users to upgrade
  their nodes to a release (such as Bitcoin Core 0.21.1) that will
  enforce taproot's rules, ensuring that funds received to taproot
  scripts after block 709,632 are safe even if there's a problem with
  miners.

    Developers are encouraged to start [implementing taproot][taproot
    uses] in their software during the next several months so they can
    be ready to take advantage of greater efficiency, privacy, and

>     fungibility as soon as the activation is complete.

Reviewer: "start implementing [...] during the next months [...] to be ready" feels like a bit too hedged to me. 

I'd either `s/start implementing/implement/` or `s/during the next several months//`

  and increases its [PSBT][topic psbt] capabilities, among other improvements
  and bug fixes.

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
repo], [Hardware Wallet Interface (HWI)][hwi repo],
[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
BOLTs][bolts repo].*

- [Bitcoin Core GUI #4][] adds initial support for using [Hardware Wallet
  Interface (HWI)][topic hwi] external signers via the GUI. Once this feature is
  finalized, users will be able to use their HWI-compatible hardware wallets
  directly from the Bitcoin Core GUI.

- [Bitcoin Core #21573][] updates the version of libsecp256k1 included
  in Bitcoin Core.  Of the changes, the most notable in the use of the
  optimized modular inverse code described in Newsletters [#136][news136
  safegcd] and [#146][news146 safegcd].  Performance evaluations posted
  to the PR found it to speed up old block verification by about 10%.

> - [C-Lightning #4591][] adds support for parsing bech32m addresses. C-Lightning

Author: ```suggestion
- [C-Lightning #4591][] adds support for parsing [bech32m][topic bech32] addresses. C-Lightning
```

Reviewer: Done!

  will now admit creation of Pay to Taproot (P2TR) (and other v1+ native
  segwit) outputs when `option_shutdown_anysegwit` is set. Note that P2TR

>   outputs are non-standard and `ANYONE_CAN_SPEND` on mainnet before the Taproot

Author: Txes with P2TR outputs **are** standard (that's the whole point of this PR).  Txes with P2TR *inputs* are non-standard.  Also I think your formating is confusing anyone-can-spend (lowercase, just a phrase) with `SIGHASH_ANYONECANPAY`.  Instead of saying "anyone can pay", I usually try to be a bit more explicit and say something like, "before the taproot soft fork activates in November, payments sent to a taproot address or any future segwit version address have no security---anyone can spend them."

P.S.  :rofl: youl comment in the commit message.

Reviewer: Ah rats, I misunderstood the option to be a configuration option for the node. I hope this is better.
